pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -155,6 +156,8 @@ func (rc *reconciler) reconcile() {
 
 			// Before triggering volume detach, mark volume as detached and update the node status
 			// If it fails to update node status, skip detach volume
+			// TODO-P1: check error code
+			// TODO: Not sure I understand why we remove it first
```

Comment:
This is to make sure no race between controller detach operation on master and mount operation by kubelet on node. Considering the following case,
1. Master attach volume and update it in node status showing that the volume is attached.
2. On node, controller verify the node is attached, and continue to mount the volume
3. Master checks it is safe to detach, and continue to detach.
Now between 2 and 3, there is a chance that controller is trying to detach while the volume is mounted.
To avoid such race condition, both master and node try to update the status before continuing the volume operation.
Some details are in issue #40887